pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -471,11 +482,23 @@ func (n *NodeInfo) Clone() *NodeInfo {
 // VolumeLimits returns volume limits associated with the node
 func (n *NodeInfo) VolumeLimits() map[v1.ResourceName]int64 {
 	volumeLimits := map[v1.ResourceName]int64{}
+
 	for k, v := range n.AllocatableResource().ScalarResources {
 		if v1helper.IsAttachableVolumeResourceName(k) {
 			volumeLimits[k] = v
 		}
 	}
+
+	if n.csiNode != nil {
+		for i := range n.csiNode.Spec.Drivers {
+			d := n.csiNode.Spec.Drivers[i]
+			if d.Allocatable != nil && d.Allocatable.Count != nil {
+				k := v1.ResourceName(volumeutil.GetCSIAttachLimitKey(d.Name))
```

Comment:
Should we be calling GetCSIAttachLimitKey()? It does the hashing thing when driver name is too big, which the new design should handle.  We should also make sure that non-migration CSI drivers will use the correct resource name.

Reply:
I think this is to ensure that, both new driver names coming from `CSINode` and old names coming from `Node` object are consolidated with unique keys in the `volumeLimits` map. Otherwise, we will end up with 2 values for migrated plugins. 

As a interim step I think it is fine until we can stop recognizing limits coming from node objects..

Reply:
K let's just add a comment to clean this up later

Reply:
Added TODO